collector.go

```diff
@@ -0,0 +1,59 @@
+package versionbundle
+
+import (
+	"net/url"
+
+	"github.com/giantswarm/microerror"
+	"github.com/go-resty/resty"
+)
+
+type CollectorConfig struct {
+	RestClient *resty.Client
+
+	Endpoints []url.URL
+}
+
+func DefaultCollectorConfig() CollectorConfig {
+	return CollectorConfig{
+		RestClient: nil,
+
+		Endpoints: nil,
+	}
+}
+
+type Collector struct {
+	restClient *resty.Client
+
+	bundles []Bundle
+
+	endpoints []url.URL
+}
+
+func NewCollector(config CollectorConfig) (*Collector, error) {
+	if config.RestClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.RestClient must not be empty")
+	}
+
+	if len(config.Endpoints) == 0 {
+		return nil, microerror.Maskf(invalidConfigError, "config.Endpoints must not be empty")
+	}
+
+	c := &Collector{
+		restClient: config.RestClient,
+
+		bundles: nil,
+
+		endpoints: config.Endpoints,
+	}
+
+	return c, nil
+}
+
+func (c *Collector) Bundles() []Bundle {
+	return c.bundles
+}
```

Comment:
The idea is to have a dumb getter to return the cached bundles found so far. The flat list of bundles contains any bundles being exposed by any of the configured endpoints. The result provided by the collector can then be given to `versionbundle.Aggregate`. The result of this in turn can be used to compute distribution types we introduced in a former PR.